edkrepo/common/common_repo_functions.py

```diff
@@ -913,6 +893,33 @@ def create_local_branch(name, patchset, global_manifest_path, manifest_obj, repo
             json.dump(data, f, indent=4)
         f.close()
 
+def _list_patchset_ancestors(manifest, patchset):
+    parent_patchsets = []
+    try:
+        parent_patchsets.append(manifest.get_patchset(patchset.parent_sha, patchset.remote))
+        while True:
+            # Allow daisy-chaining PatchSet.  Continue until KeyError is caught.
+            parent_patchsets.append(manifest.get_patchset(parent_patchsets[-1].parent_sha, parent_patchsets[-1].remote))
+            if parent_patchsets.count(parent_patchsets[-1]) > 1:
+                # Do not continue to branch creation step if a circular dependency is detected between multiple PatchSet.
+                raise ValueError(PATCHSET_CIRCULAR_DEPENDENCY_ERROR.format(parent_patchsets[-1]))
+    except KeyError:
+        return parent_patchsets
+
+def _checkout_parent(patchset, name, repo, parent_patchsets):

```

Comment:
def _checkout_parent(patchset, name, repo, parent_patchsets):
    if parent_patchsets:
        base_parent_sha = parent_patchsets[-1].parent_sha
        if (base_parent_sha in repo.heads and
            base_parent_sha not in repo.tags):
            # PatchSet should not use a branch name as the base parent_sha.
            print("PatchSet should not use branch name as base parent_sha")
    else:
        base_parent_sha = patchset.parent_sha

    if base_parent_sha in repo.tags:
        # tag names are prioritized over branch names
        repo.git.checkout("refs/tags/{}".format(base_parent_sha), b=name)
    else:
        repo.git.checkout(base_parent_sha, b=name)

Reply:
@ndhaller Should there be any exception raised after the print statement, or is it more of a warning?

Reply:
For the open issue 139 is should be an exception.  I haven't written the exception yet since I saw you are refactoring that code.  
https://github.com/tianocore/edk2-edkrepo/issues/139

Reply:
Since this exception and refactoring are more related to a different issue, I've applied these changes and opened a separate PR at https://github.com/tianocore/edk2-edkrepo/pull/159, with credit of course. Thank you!